### PR TITLE
SSE Loss for XGBoost

### PR DESCRIPTION
**Pull request by RishabhGarg108**
This PR attempts to add SSE (Sum of Squared Errors) Loss function for xgboost. The list of methods implemented as of now is:-
1. InitialPrediction - Gives the initial constant prediction used in boosting.
2. Gradients - Calculates the first-order gradients of the loss function. (Both for scalars and vectors.)
3. Hessians - Calculates the second-order gradients of the loss function. (Both for scalars and vectors.)
4. Residuals - Calculates the pseudo residuals of the predictions.

Methods yet to be implemented:-
1. EvaluateGain - Calculates gain for splitting the tree.
2. OutputValue - Calculates the output value for the leaves of the decision tree.

The implementation which I will be doing in this PR will be pretty simple and without any optimizations. Gradually we will keep iterating over things and improving them one at a time.

Also, the methods which are implemented or yet to be implemented (mentioned above) are the basic ones that will be required very frequently. We will keep adding more methods if required later.

CC - @gmanlan @rcurtin

### CODE REVIEW

**Comment by RishabhGarg108** [comment]
Oops. I created this branch from the regression-tree branch and the commits from it are also showing in this. I will probably create another PR once this is completed to fix this commits issue.

**Comment by RishabhGarg108** [comment]
Closing in favour of #3003.
